### ISSUE TITLE
vimb is crashing when visiting https://nextjs.org

### ISSUE THREAD

**Issue by yashasolutions** [fanglingsu/vimb#655]
### Steps to reproduce

`$ vimb https://nextjs.org/`

### Expected behaviour

Loading the site

### Actual behaviour

Error message at bottom `webview crashed on https://nextjs.org/`

### Debug info

```
Version:         3.6.0-24-gc626ab2
WebKit compile:  2.24.2
WebKit run:      2.24.2
GTK compile:     3.24.4
GTK run:         3.24.4
libsoup compile: 2.64.1
libsoup run:     2.64.1
Extension dir:   /usr/local/lib/vimb
```

### Possible reasons for error

I have the same issue with surf (who's also based on webkit) 
Could be the version of webkit that need upgrade.
Same site works fine in FF68, or Chrome

**Comment by fanglingsu** [comment]
@yashasolutions I can not reproduce the crash with
```
Version:         3.6.0-24-gc626ab2
WebKit compile:  2.30.4
WebKit run:      2.30.4
GTK compile:     3.24.24
GTK run:         3.24.24
libsoup compile: 2.72.0
libsoup run:     2.72.0
Extension dir:   /usr/lib/vimb
```
So it might be related to the webkit version. If you run vimb from CLI you might get additional information about the reason or you might get a coredumb of the crashed webextension.

**Comment by jjakob** [comment]
I and another user had a problem on Gentoo with the WebkitWebProcess crashing on various sites that was reported in a webkit bug: https://bugs.webkit.org/show_bug.cgi?id=222358
There is a workaround in the bug report that works great, by just exporting `JavaScriptCoreUseJIT=0` or `JSC_useBaselineJIT=false`. I've added that to my .bashrc and I have no more crashing webkit since then. Maybe your issue is the same.

**Comment by jjakob** [comment]
That bug was marked as a duplicate of https://bugs.webkit.org/show_bug.cgi?id=222805
It seems like it's caused by webkit-gtk javascript JIT needing the SSE4 instruction that isn't present on older CPUs.

**Comment by fennewald** [comment]
I'm getting this issue with linkedin.com, most recent webkit installed. Here's my bug-info:
```Version:         3.6.0-34-g652f411
WebKit compile:  2.32.1
WebKit run:      2.32.1
GTK compile:     3.24.29
GTK run:         3.24.29
libsoup compile: 2.72.0
libsoup run:     2.72.0
Extension dir:   /usr/local/lib/vimb```

Setting `JavaScriptCoreUseJIT` or `JSC_useBaselineJIT` does not help.  When I spawn vimb from the console, I get the following error: `(vimb-raw:8634): GLib-CRITICAL **: 14:50:25.263: Source ID 46 was not found when attempting to remove it`

This seems like it may be a seperate issue. If asked, I will open up a seperate issue.

**Comment by jjakob** [comment]
Are you using Gentoo? I tried 2.32.1 but had freezing issues with it.
It didn't crash but just froze on pages. I'd say it's a WebKit issue,
as 2.30.6 works fine with the same version of vimb.

**Comment by fennewald** [comment]
@jjakob nope, I'm on void linux, but I didn't install vimb through any package manager, just did it from source
